lib/index.js

```diff
@@ -85,31 +85,57 @@ const fixupNestedPseudo = astNode => {
   transformAst(newRootNode)
 }
 
-// :semver(<version|range>, [selector], [function])
+// :semver(<version|range|selector>, [version|range|selector], [function])
+// note: the first or second parameter must be a static version or range
 const fixupSemverSpecs = astNode => {
-  // the first child node contains the version or range, most likely as a tag and a series of
-  // classes. we combine them into a single string here. this is the only required input.
-  const children = astNode.nodes.shift().nodes
-  const value = children.reduce((res, i) => `${res}${String(i)}`, '')
-
-  // next, if we have 2 nodes left then the user called us with a total of 3. that means the
-  // last one tells us what specific semver function the user is requesting, so we pull that out
-  let semverFunc
-  if (astNode.nodes.length === 2) {
+  // if we have three nodes, the last is the semver function to use, pull that out first
+  if (astNode.nodes.length === 3) {
     const funcNode = astNode.nodes.pop().nodes[0]
     if (funcNode.type === 'tag') {
-      semverFunc = funcNode.value
+      astNode.semverFunc = funcNode.value
+    } else if (funcNode.type === 'string') {
+      // a string is always in some type of quotes, we don't want those so slice them off
+      astNode.semverFunc = funcNode.value.slice(1, -1)
+    } else {
+      // anything that isn't a tag or a string isn't a function
+      throw Object.assign(
+        new Error('`:semver` pseudo-class expects a function name as last value'),
+        { code: 'ESEMVERFUNC' }
+      )
+    }
+  }
+
+  // now if we have 1 node, it's a static value
+  // istanbul ignore else
```

Comment:
what's the else state here?

Reply:
the else here would be if more than 3 arguments are provided to the selector, which should perhaps throw right at the top of this function. i didn't see that behavior elsewhere (throwing if too many args are provided) so i didn't implement it here, but i certainly could